src/Pages/MainPage.js

```diff
@@ -0,0 +1,118 @@
+import React, {useEffect, useState} from 'react';
+import axios from 'axios';
+import {Route, Switch, useParams, useHistory} from "react-router-dom";
+import './MainPage.css';
+import {PokemonCard} from "../Components/Molecules/PokemonCard";
+
+function MainPage() {
+
+    const history = useHistory();
+
+    //get offset from url, force it to be an integer and catch
+    //some possible erroneous input
+    let {offset} = useParams();
+    offset = parseInt(offset);
+    if (offset <= 0) {
+        offset = 0;
+        history.push('/')
+    }
+    if (!offset) {
+        offset = 0;
+    }
+
+    const [data, setData] = useState(null);
+    const [error, toggleError] = useState(false);
+    const [hasNext, toggleHasNext] = useState(true);
+    const [hasPrevious, toggleHasPrevious] = useState(false);
+
+
+    useEffect(() => {
+
+            const cancelTokenSource = axios.CancelToken.source();
+            async function fetchData() {
+                try {
+                    const result = await axios.get(`https://pokeapi.co/api/v2/pokemon?offset=${offset}&limit=20`, {cancelToken: cancelTokenSource.token});
+                    setData(result.data);
+                    toggleError(false)
+                    if (result.data.previous === null) {
+                        toggleHasPrevious(false);
+                    } else {
+                        toggleHasPrevious(true);
+                    }
+                    if (result.data.next === null) {
+                        toggleHasNext(false);
+                    } else {
+                        toggleHasNext(true);
```

Comment:
Ik denk dat je dit korter kunt opschrijven, dan heb je deze hele beslisboom niet meer nodig :) `null` wordt gezien als `falsy` namelijk. Als er een waarde in zit, zal de vergelijking true geven, als er null in zit, zal de vergelijking `result.data.previous === true` **niet** waar zijn

`toggleHasPrevious(result.data.previous === true);`
`toggleHasNext(result.data.next === true);`